contracts/abstract/StrategyInvestor.sol

```diff
@@ -0,0 +1,407 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.26;
+
+import {IERC20Upgradeable, SafeERC20Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import {IBeefyVaultV7} from '../interfaces/IBeefyVaultV7.sol';
+import {VaultManager} from '../VaultManager.sol';
+import {LiquidityManager} from "../LiquidityManager.sol";
+import {ZapManager} from "../zap/ZapManager.sol";
+import {ZapLib} from "../libraries/ZapLib.sol";
+import {StrategyStorage} from "./StrategyStorage.sol";
+import {SubscriptionManager} from "../SubscriptionManager.sol";
+import {UseFee} from "./UseFee.sol";
+
+contract StrategyInvestor is StrategyStorage {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    error InvalidParamsLength();
+    error InsufficientFunds();
+
+    struct DcaInvestmentParams {
+        DcaInvestment[] dcaInvestments;
+        IERC20Upgradeable inputToken;
+        uint amount;
+        bytes[] swaps;
+    }
+
+    struct VaultInvestmentParams {
+        VaultInvestment[] vaultInvestments;
+        IERC20Upgradeable inputToken;
+        uint amount;
+        bytes[] swaps;
+    }
+
+    /// part of LiquidityInvestParams
+    struct LiquidityInvestZapParams {
+        bytes swapToken0;
+        bytes swapToken1;
+        uint swapAmountToken0;
+        uint swapAmountToken1;
+        int24 tickLower;
+        int24 tickUpper;
+        uint amount0Min;
+        uint amount1Min;
+    }
+
+    struct LiquidityInvestParams {
```

Comment:
I'd suggest following the same naming pattern for all structs. If you take a look at the DCA and Vault structs, they are named with the `Investment` suffix, but Liquidity and Buy use the `Invest`...